sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ArrayBasedMapData.scala

```diff
@@ -136,7 +136,8 @@ object ArrayBasedMapData {
     keys.zip(values).toMap
   }
 
-  def toScalaMap(keys: Seq[Any], values: Seq[Any]): Map[Any, Any] = {
+  def toScalaMap(keys: scala.collection.Seq[Any],
```

Comment:
Does this has to accept s.c.Seq?

Reply:
Yep, now it must be `s.c.Seq` unless we accept `case ArrayType(et, nullable)` of `RowEncoder#deserializerFor` method has a 2.13-only branch to use `immutable.ArraySeq.unsafeMake` create a `immutable.Seq`, can't use uniformly` immutable.ArraySeq ` because they are in different packages with different Scala version,  I haven't found a replacement of `WrappedArray.make` method yet.

Now `WrappedArray.make` create a `mutable.ArraySeq` instance in Scala 2.13 , so  the `toScalaMap` method input args are `mutable.ArraySeq` type not `immutable.ArraySeq` and `Seq` alias representative ` `immutable.Seq`, this is not a match.

On the other hand the `toScalaMap` method is called in reflection mode, inconvenient to add 'toSeq'.

Reply:
I see, this is called by things that need to pattern-match s.c.Seq.